src/sensor.h

```diff
@@ -90,6 +91,8 @@ namespace librealsense
             return {};
         }
 
+        virtual bool supports_feature( const std::string & feature_name ) const { return false; } // Default implementation, override in derived classes
```

Comment:
I'm confused.
I thought it was the device that supported features...?

Reply:
> No `has_features` abstract interface?
If we will want in the future we can add a `get_all_features` API for feature serialization.

> We decided to go with a name identifier?
I think that name is more searchable then GUID and I don't currently see need for both.

Reply:
> In the meeting we talked about changing the control level to `sensor`. I agreed with you that different sensors of the same device might or might not have support for the same feature type.

So what about device features?

Reply:
Added const IDs to features.
Don't think that we need device level features at this point.
We are not trying to replace the options mechanism, just create a standardized way of inquiring if we need to register options (or other initializations) based on run time decision.

Reply:
Wouldn't you want a `has_features` abstract class that defines what a class that has features needs to implement? That can be reused by `device_interface` when we decide to add it there?

Reply:
Implemented `features_support` abstract and `features_container` implementation that devices can use.